pg_search/src/postgres/customscan/aggregatescan/scan_state.rs

```diff
@@ -128,7 +128,42 @@ impl AggregateScanState {
             }
         };
 
-        // Collect filter results for lookup
+        // Check if this is direct format (numeric keys) or filter format (filter_* keys)
+        let has_direct_keys = result_obj.keys().any(|k| k.parse::<usize>().is_ok());
+
+        if has_direct_keys {
```

Comment:
This reverse-engineering of key types is awkward. Can this information be passed in some other way?

But also: this seems like a separate optimization, unrelated to fixing the regression. Can/should we do it in a separate PR, post cleanup?